api/internal/localizer/localizer_test.go

```diff
@@ -0,0 +1,468 @@
+// Copyright 2022 The Kubernetes Authors.
```

Comment:
The localizer.go looks good to me! Now we need some unittests to cover the `NewLocalizer` and `Localize` methods. 
I'm ok with using some helper functions in _test.go like `RunLocalize`, but the unittests should cover the code (right now they are skipped).

- remove skipped unittests (we should add them back once the repo path is fixed)
- add the following unittests
   1.    call `NewLocalizer`, pass with the `Localizer` attributes initialized, related dir created.
   2.    (optional) call `NewLocalizer` that triggers error "cannot find path from directory %q to %q inside directory: %s"
   3.  call `Localize` that can load, parse and write a basic kustomization content to local

For 3,
you would need some mocked processKust 

 In localizer.go
```
// TODO: test only, will be replaced by Localizer.processKust
var processKustFn = processKust

// func (lc *Localizer) processKust(_ *target.KustTarget) (*types.Kustomization, error) {
func processKust(_ *Localizer, _ *target.KustTarget) (*types.Kustomization, error) {}
```

 In localizer_test.go
```
func TestLocalize(..) {
     ...
     processKustFn = fakeProcessKust
}

// Now you can customize the behavior of  fakeProcessKust.
func fakeProcessKust()
```

Reply:
Thank you so much for this suggestion